xdsl/dialects/riscv.py

```diff
@@ -331,6 +334,73 @@ class RISCVOp(Operation, ABC):
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        operands = cls.parse_operands(parser)
+        successors = cls.parse_successors(parser)
+        regions = cls.parse_regions(parser)
```

Comment:
I would remove successors and regions until they are needed. Only a very small subset of RISC-V Operations actually require them.

Reply:
If that is possible that would be great. Did you test if this is indeed dead code?